File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/tufilm/KinoAdapter.java
package de.tum.in.tumcampusapp.component.ui.tufilm;

import java.util.List;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;
import de.tum.in.tumcampusapp.component.ui.tufilm.details.KinoDetailsFragment;
import de.tum.in.tumcampusapp.component.ui.tufilm.model.Kino;

public class KinoAdapter extends FragmentStatePagerAdapter {

    private final List<Kino> movies;

    KinoAdapter(FragmentManager fm, List<Kino> kinos) {
        super(fm);
        movies = kinos;
    }

    @Override
    public Fragment getItem(int position) {
        String id = movies.get(position).getId();
        return KinoDetailsFragment.newInstance(id);
    }

    @Override
    public int getCount() {
        return movies.size();
    }

}

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/tufilm/KinoAdapter.kt
package de.tum.`in`.tumcampusapp.component.ui.tufilm

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

import de.tum.`in`.tumcampusapp.component.ui.tufilm.model.Kino

class KinoAdapter internal constructor(
        fragmentManager: FragmentManager, private val movies: List<Kino>
) : FragmentStatePagerAdapter(fragmentManager) {

    override fun getItem(position: Int): Fragment = KinoDetailsFragment.newInstance(position)

    override fun getCount() = movies.size

}